library/core/src/option.rs

```diff
@@ -2545,3 +2546,41 @@ impl<T> Option<Option<T>> {
         }
     }
 }
+
+impl<T, const N: usize> [Option<T>; N] {
+    /// Transposes a `[Option<T>; N]` into a `Option<[T; N]>`.
+    ///
```

Comment:
Maybe something to explain the behavior?

```rust
    /// Transposes a `[Option<T>; N]` into a `Option<[T; N]>`.
    ///
    /// This will return `Some` only if every input element is `Some`. If any element is `None`,
    /// all values are dropped and this returns `None`.
```